azure/application-gateway/advanced/examples/advanced/main.tf

```diff
@@ -0,0 +1,151 @@
+provider "azurerm" {
+  features {
+    resource_group {
+      prevent_deletion_if_contains_resources = false
+    }
+  }
+}
+
+locals {
+  tags = {
+    module  = "application-gateway"
+    example = "advanced"
+    usage   = "demo"
+  }
+}
+
+resource "azurerm_resource_group" "example" {
+  name     = "awg-rg"
+  location = "uksouth"
+}
+
+resource "azurerm_virtual_network" "example" {
+  name                = "vnet"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  tags                = local.tags
+
+  address_space = ["10.0.0.0/24"]
+}
+
+resource "azurerm_subnet" "example" {
+  name                 = "default"
+  resource_group_name  = azurerm_resource_group.example.name
+  virtual_network_name = azurerm_virtual_network.example.name
+
+  address_prefixes = azurerm_virtual_network.example.address_space
+}
+
+resource "azurerm_user_assigned_identity" "example" {
+  name                = "umid-agw"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+}
+
+module "example" {
+  source = "../../"
+
+  name                = "agw-example"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  tags                = local.tags
+  subnet_id           = azurerm_subnet.example.id
+
+  sku = {
+    name         = "WAF_v2"
+    tier         = "WAF_v2"
+    capacity     = 1
+    max_capacity = 3
+  }
+
+  identity_ids = [azurerm_user_assigned_identity.example.id]
+
+  private_ip       = cidrhost(one(azurerm_subnet.example.address_prefixes), 5)
+  create_public_ip = true
+  pip_name         = "agw-pip"
+
+  additional_frontend_ports = {
+    alt_http = 8080
+  }
+
+  enable_http2 = true
+
+  ssl_certificates = {
+    example_cert = {
+      data     = filebase64("selfsigned.pfx")
```

Comment:
Should the cert be in it's own dir like 'files'?

Reply:
Perhaps but it's only an example so it doesn't matter all that much.

it's more to highlight "this is what you can do with the module"

Ideally, we wouldn't store a certificate in Git at all and we'd put it in Keyvault instead